### ISSUE TITLE
Function tags conflict with class specifiers in `forward` declarations

### ISSUE THREAD

**Issue by Daniel-Cortez** [Zeex/pawn#635]
### Issue description:

The following code compiles correctly:
```Pawn
forward public Func();
```
But when specifying a tag for the return value, it won't compile
```Pawn
forward public Tag:Func(); // error 010: invalid function or declaration
```
, although if the class specifier is removed, it would compile just fine:
```Pawn
forward Tag:Func(); // no errors
```

### Minimal complete verifiable example (MCVE):

See above.

### Workspace Information:

* Compiler version: 3.10.10, 3.2.3664
* Command line arguments provided (or sampctl version):
* Operating System:

**Comment by Daniel-Cortez** [comment]
Checked the code in function `funcstub()` (file `sc1.c`); apparently the compiler expects the tag *before* the class specifier (which is obviously a bug), so, for example, this code
```Pawn
forward Tag:public Func();
```
surprisingly compiles.

**Comment by BitFros7y** [comment]
Possibly related to https://github.com/pawn-lang/compiler/issues/618 ?

**Comment by Daniel-Cortez** [comment]
> Possibly related to #618 ?

No, that one is related to function arguments, while this bug affects the tag of the function itself. Also, what you described in your issue is intended behavior (although I agree it might seem counter-intuitive at times) while this one clearly isn't.
